test/test_oneview_power_device.py

```diff
@@ -0,0 +1,431 @@
+
+        PowerDeviceModule().run()
+
+        mock_ansible_instance.exit_json.assert_called_once_with(
+            changed=True,
+            msg=POWER_DEVICE_UPDATED,
+            ansible_facts=dict(power_device={'name': 'name'})
+        )
+
+    @mock.patch.object(OneViewClient, 'from_json_file')
+    @mock.patch('oneview_power_device.AnsibleModule')
+    def test_should_fail_when_same_name_already_exists(self, mock_ansible_module, mock_ov_client_from_json_file):
+        mock_ov_instance = mock.Mock()
+        mock_ov_instance.power_devices.get_by.return_value = [DICT_DEFAULT_POWER_DEVICE]
+
+        mock_ov_client_from_json_file.return_value = mock_ov_instance
+        mock_ansible_instance = create_ansible_mock(YAML_POWER_DEVICE_CHANGE)
+        mock_ansible_module.return_value = mock_ansible_instance
+        mock_ansible_instance.exit_json.side_effect = sys.exit
+
+        PowerDeviceModule().run()
+
+        mock_ansible_instance.exit_json.assert_called_once_with(
+            changed=False,
+            msg=POWER_DEVICE_NEW_NAME_ALERADY_EXISTS
+        )
+
+    @mock.patch.object(OneViewClient, 'from_json_file')
+    @mock.patch('oneview_power_device.AnsibleModule')
+    def test_should_fail_when_power_device_not_exists(self, mock_ansible_module, mock_ov_client_from_json_file):
+        mock_ov_instance = mock.Mock()
+        mock_ov_instance.power_devices.get_by.return_value = []
+
+        mock_ov_client_from_json_file.return_value = mock_ov_instance
+        mock_ansible_instance = create_ansible_mock(YAML_POWER_DEVICE_CHANGE)
+        mock_ansible_module.return_value = mock_ansible_instance
+        mock_ansible_instance.exit_json.side_effect = sys.exit
+
+        PowerDeviceModule().run()
+
+        mock_ansible_instance.exit_json.assert_called_once_with(
+            changed=False,
+            msg=POWER_DEVICE_NOT_FOUND
+        )
+
+    @mock.patch.object(OneViewClient, 'from_json_file')
+    @mock.patch('oneview_power_device.AnsibleModule')
+    def test_should_fail_with_missing_required_attributes(self, mock_ansible_module, mock_ov_client_from_json_file):
+        mock_ov_instance = mock.Mock()
+
+        mock_ov_client_from_json_file.return_value = mock_ov_instance
+
+        mock_ansible_instance = mock.Mock()
+        mock_ansible_instance.params = {"state": "present",
+                                        "config": "config",
+                                        "data":
+                                            {"field": "invalid"}}
+        mock_ansible_module.return_value = mock_ansible_instance
+
+        PowerDeviceModule().run()
+
+        mock_ansible_instance.fail_json.assert_called_once_with(
+            msg=POWER_DEVICE_MANDATORY_FIELD_MISSING
+        )
+
+    @mock.patch.object(OneViewClient, 'from_json_file')
+    @mock.patch('oneview_power_device.AnsibleModule')
+    def test_should_not_add_when_add_raises_exception(self, mock_ansible_module, mock_ov_client_from_json_file):
```

Comment:
I suggest change the name of the method to `test_should_fail_when_add_raises_exception`